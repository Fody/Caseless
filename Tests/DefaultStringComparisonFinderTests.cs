using System.Xml.Linq;
using NUnit.Framework;

[TestFixture]
public class DefaultStringComparisonFinderTests
{
    [Test]
    public void GetStringComparisonFromXml()
    {
        var xElement = XElement.Parse("<Caseless StringComparison='InvariantCultureIgnoreCase'/>");
        Assert.AreEqual("InvariantCultureIgnoreCase", DefaultStringComparisonFinder.GetStringComparisonFromXml(xElement));
    }

    [Test]
    public void GetStringComparisonFromXmlNull()
    {
        DefaultStringComparisonFinder.GetStringComparisonFromXml(null);
    }

    [Test]
    public void GetStringComparisonFromXmlTrim()
    {
        var xElement = XElement.Parse("<Caseless StringComparison=' InvariantCultureIgnoreCase '/>");
        Assert.AreEqual("InvariantCultureIgnoreCase", DefaultStringComparisonFinder.GetStringComparisonFromXml(xElement));
    }

    [Test]
    public void GetStringComparisonFromXmlWhiteSpace()
    {
        var xElement = XElement.Parse("<Caseless StringComparison='  '/>");
        var exception = Assert.Throws<WeavingException>(() => DefaultStringComparisonFinder.GetStringComparisonFromXml(xElement));
        Assert.AreEqual("Expected StringComparison to have a value.", exception.Message);
    }

    [Test]
    public void GetStringComparisonFromXmlEmpty()
    {
        var xElement = XElement.Parse("<Caseless StringComparison=''/>");
        var exception = Assert.Throws<WeavingException>(() => DefaultStringComparisonFinder.GetStringComparisonFromXml(xElement));
        Assert.AreEqual("Expected StringComparison to have a value.", exception.Message);
    }
}