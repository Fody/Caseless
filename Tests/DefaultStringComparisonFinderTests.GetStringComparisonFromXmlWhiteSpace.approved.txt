Expected StringComparison to have a value.